src/components/download/helperFunctions.js

```diff
@@ -623,15 +623,19 @@ export const entropyTSV = (dispatch, filePrefix, entropy) => {
 /**
  * Write out Auspice JSON(s) for the current view. We do this by re-fetching the original
  * JSON because we don't keep a copy of the unprocessed data around.
- * 
+ *
  * Sidecar files are not fetched, but we can also download them if desired.
- * 
+ *
  * Note that we are not viewing a narrative, as the download button functionality is disabled
  * for narratives.
  */
-export const auspiceJSON = (dispatch) => {
+export const auspiceJSON = (dispatch, datasetNames) => {
   const filenames = [];
-  for (const datasetName of getDatasetNamesFromUrl(window.location.pathname)) {
+  if (datasetNames.every((x) => x === '')) {
+    console.error("Unable to fetch empty dataset names");
```

Comment:
I'm totally unfamiliar with this code, but I wonder if logging the value of `dispatch` here would be helpful in figuring out what's going on, if this error is ever triggered? (I realize it's a "this should never happen" category thing.)
 
```suggestion
    console.error(`Unable to fetch empty dataset names for ${dispatch}`);
```

Reply:
Ah, `dispatch` is the [Redux dispatch](https://redux.js.org/api/store#dispatchaction) for dispatching the error notification below so it would not be meaningful here.

Reply:
I ended up needing to see what the `datasetNames` value was in my debugging, so I'm logging out `${JSON.stringify(datasetNames)}`  in https://github.com/nextstrain/auspice/pull/1804/commits/45bd85d4cca351ab5447c02d507b8dba69cdce05